src/mame/quantel/dpb7000.cpp

```diff
@@ -1821,35 +2423,176 @@ void dpb7000_state::cpu_ctrlbus_w(uint16_t data)
 
 	case 15: // Disk Sequencer Card, panic reset
 		LOGMASKED(LOG_CTRLBUS, "%s: CPU write to Control Bus, Disk Sequencer Card, panic reset\n", machine().describe_context());
-		m_diskseq_reset = true;
-		m_diskseq_halt = true;
 		break;
+
 	default:
 		LOGMASKED(LOG_CTRLBUS | LOG_UNKNOWN, "%s: CPU write to Control Bus, unknown CSR %d: %04x\n", machine().describe_context(), m_csr, data);
 		break;
 	}
 }
 
-WRITE_LINE_MEMBER(dpb7000_state::req_a_w)
+TIMER_CALLBACK_MEMBER(dpb7000_state::req_a_w)
 {
-	if (state)
-		m_sys_ctrl |= SYSCTRL_REQ_A_IN;
-	else
-		m_sys_ctrl &= ~SYSCTRL_REQ_A_IN;
+	//if (machine().input().code_pressed(KEYCODE_LALT))
+	{
+		if (param)
+		{
+			m_sys_ctrl |= SYSCTRL_REQ_A_IN;
+		}
+		else
+		{
+			m_sys_ctrl &= ~SYSCTRL_REQ_A_IN;
+		}
 
-	update_req_irqs();
+		update_req_irqs();
+	}
+	/*else
+	{
+		m_sys_ctrl &= ~SYSCTRL_REQ_A_IN;
+		update_req_irqs();
+	}*/
 }
 
-WRITE_LINE_MEMBER(dpb7000_state::req_b_w)
+TIMER_CALLBACK_MEMBER(dpb7000_state::req_b_w)
 {
-	if (state)
+	if (param)
+	{
 		m_sys_ctrl |= SYSCTRL_REQ_B_IN;
+	}
 	else
+	{
 		m_sys_ctrl &= ~SYSCTRL_REQ_B_IN;
+	}
 
 	update_req_irqs();
 }
 
+TIMER_CALLBACK_MEMBER(dpb7000_state::cmd_done)
+{
+	m_brush_addr_func &= ~0x8000;
+}
+
+TIMER_CALLBACK_MEMBER(dpb7000_state::execute_hdd_command)
+{
+	static const int s_sectors_per_track = 20480 / 256;
```

Comment:
Since this is a constant and not really a static, I’d prefer it as `constexpr SECTORS_PER_TRACK = ...`

The `static` isn’t even really necessary.  The compiler won’t allocate storage for it if it knows it can use the value directly.  The only difference adding `static` makes is that if you take the address of it for some reason, the compiler needs to ensure it always has the same address.  That isn’t necessary since pointers to it won’t escape an invocation of the function.

Reply:
Good call, switched all non-string constants to use constexpr and `SCREAM_CASE`.